### ISSUE TITLE
no longer working in Atom

### ISSUE THREAD

**Issue by Orebeer** [imdone/imdone-atom#288]
Currently using Atom ver `1.23.1` and it no longer works. 

I used to have many cards, now it only pops up one within one "DONE" column. Everything else is gone. the config.js also doesn't sync with the labeling conventions on the columns or filters which would autocompile from within the documentation. 

Tried replicating the issue in new repos and projects and its the same error: always one card within the "DONE" column. 

Tried uninstalling and reinstalling and wiping the config.js within the .imdone dir. Same thing. Nothing works.

**Comment by piascikj** [comment]
What version of imdone-atom?  Can you include a sample file who's TODOs are not showing up?  Are you using .imdoneignore?

**Comment by Orebeer** [comment]
#### Setup
- It's the most recent: imdone-atom version `2.3.18` 
- not using .imdoneignore
- there is nothing manually added within the config.js "exclude" array outside of its defaults:

```
  "exclude": [
    "^(node_modules|bower_components|\\.imdone|target|build|dist|logs|flow-typed)[\\/\\\\]?|\\.(git|svn|hg|npmignore)|\\~$|\\.(jpg|png|gif|swp|ttf|otf)$"
  ],

```
- Vcs paths are not excluded
- Alternate File Watcher does not help
- have tried setting up a new project in Atom to QA the issue:

   - Used a new project folder that is not a sub-dir of any parent folders holding .imdone folders.
   - When setting up a new project, I use 3 default template markdown files to build the columns and filters within the imdone Kanban

---

#### What's interesting to note
The only column it will now adds is the "Done" column. Usually it will build all from this Columns.md template:
```
#BACKLOG:100_
#MUST:100_
#SHOULD:100_
#COULD:100_
#WONT:100_
#DONE:100_
#GUIDANCE:100_
```
Ordinarily that would immediately build those columns with a single card labeled "_"

---

#### config.js 

Here is the default config.json when setting up the new QA project to test this issue:

```
{
  "exclude": [
    "^(node_modules|bower_components|\\.imdone|target|build|dist|logs|flow-typed)[\\/\\\\]?|\\.(git|svn|hg|npmignore)|\\~$|\\.(jpg|png|gif|swp|ttf|otf)$"
  ],
  "watcher": true,
  "keepEmptyPriority": false,
  "code": {
    "include_lists": [
      "TODO",
      "DOING",
      "DONE",
      "PLANNING",
      "FIXME",
      "ARCHIVE",
      "HACK",
      "CHANGED",
      "XXX",
      "IDEA",
      "NOTE",
      "REVIEW",
      "DONE"
    ]
  },
  "lists": [
    {
      "name": "DONE",
      "hidden": false
    }
  ],
  "marked": {
    "gfm": true,
    "tables": true,
    "breaks": false,
    "pedantic": false,
    "smartLists": true,
    "langPrefix": "language-"
  }
}
```

**Comment by Orebeer** [comment]
### UPDATE
I got the QA Project to sort of work and in the process found the associated issue:

#### CAUSE: Conflict with [outline-selection](https://atom.io/packages/outline-selection) package
Below is the stack trace and here is the [associated issue](https://github.com/mupchrch/outline-selection/issues/12):

```
TypeError: Cannot read property 'NaN' of undefined
    at TextEditorComponent.renderedScreenLineForRow (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:959:37)
    at TextEditorComponent.pixelLeftForRowAndColumn (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:2381:35)
    at TextEditorComponent.pixelPositionForScreenPosition (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:203:46)
    at HTMLElement.pixelPositionForScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-element.js:243:32)
    at _updateOutline (/Users/Work/.atom/packages/outline-selection/lib/outline-selection.js:122:43)
    at /Users/Work/.atom/packages/outline-selection/lib/outline-selection.js:25:9
    at Array.forEach (native)
    at TextEditor.observeSelections (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:844:32)
    at /Users/Work/.atom/packages/outline-selection/lib/outline-selection.js:21:14
    at Workspace.observeTextEditors (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:725:15)
    at Object.activate (/Users/Work/.atom/packages/outline-selection/lib/outline-selection.js:18:20)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app/src/package.js:255:25)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:227:38
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:99:21)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:220:32
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app/src/package.js:217:40)
    at PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:695:42)
    at config.transactAsync (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:670:36)
    at Config.module.exports.Config.transactAsync (/Applications/Atom.app/Contents/Resources/app/src/config.js:367:24)
    at PackageManager.activatePackages (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:668:23)
    at PackageManager.activate (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:647:50)
```
---

#### Steps in finding the conflict
1. input Column names manually within **config.js**  to both arrays
```
  "code": {
    "include_lists": [
```
and also:
```
  "lists": [
		{
```
2. move the hash tasks from one column to another. they will disappear on the board. however the new hash task priority and label will change in the associated document. unlike before, there is a brief GitHub logo spinning queue when it does associate the new task labels. 
3. initiate a `Window:Reload` within the Atom menu and the items will re-appear and can now be moved around freely and their associated hash task values will update on the associated files
4. However the only way to manually add column names is within the **config.js** and if you adjust the hash task feature labels to anything that doesn't pair within **config.js**, the stack trace error will appear as referenced above. 

#### SOLUTION: buh-bye [outline-selection](https://atom.io/packages/outline-selection)...

**Comment by piascikj** [comment]
I fixed this one in 2.3.19.  Try adding a task with a new list name to a non-code file like README.md. 
[Create new lists for all tasks in non-code files · imdone/imdone-core@631d1d0](https://github.com/imdone/imdone-core/commit/631d1d07b79b8e38f28125e11d87aa2a34166ac2)

**Comment by piascikj** [comment]
@orebeer I'd like to hear more about how your using imdone-atom.  Would you mind if I contact you via email?  Just [email me](mailto:jesse@imdone.io) if you have some time to chat.
